### ISSUE TITLE
More options for building/installing documentation to ease working on it

### ISSUE THREAD

**Issue by KristofKoch** [GenericMappingTools/gmt#7225]
At the moment working on the documentation is a bit tedious. Checking my changes before committing involves building and installing, check my changes, fix/improve, build and install again, … you get the idea.

**Status quo:**

* All plots are generated from scratch every time. That is quite time consuming and for the most part unnecessary when I'm just fixing links, typos, layout, etc. Sure, `--parallel` does help but still wasted time.
* No apparent way of installing just the documentation. Not really necessary to install everything (gmt executable, library, development headers and built-in data) every time. Not as time consuming as building but still wasted time.

**Desired feature:**

* Have options to build 
  * just the `.rst`
  * just the plots
  * both (default).
* Have options to install 
  * just the documentation
  * everything (default)

**Benefits:**

* Would ease/accelerate working on the documentation.

_See https://github.com/GenericMappingTools/gmt/pull/7222#discussion_r1087558281 for the original conversation._

**Comment by PaulWessel** [comment]
Yep, make sure people use or are strongly directed to use ninja over Make.

**Comment by KristofKoch** [comment]
Not 100% sure what happens at there:

```
[…]
[ 87%] Generating ../rst/source/_images/images/GMT_general_cyl.png
[…]
```

My assumption was that the plot gets generated there from scratch. That is with an existing build directory.

**Comment by KristofKoch** [comment]
@joa-quim I have no experience with `ninja` and me trying to build the docs lead nowhere so far. Would you mind sharing your workflow to build the docs with `ninja`? So far I can build gmt with 

```
% PARALLEL_LEVEL=12 cmake -GNinja ..
% ninja
```

but I'm not sure how to install it. And how do I build the docs with it?

**Comment by PaulWessel** [comment]
See the BUILDING.md file and search for Ninja and you see a poorly document section that shows you need to run (once) build with -G Ninja or something like that.

**Comment by PaulWessel** [comment]
In a brand new setup I think

**Comment by PaulWessel** [comment]
Last comment: But it is well worth it and we just need to make how to clearer on the BUILDING.md page.

**Comment by KristofKoch** [comment]
Hi @PaulWessel, thank you for your input. What I figured out now is the following commands in a fresh `build/` directory:

`PARALLEL_LEVEL=12 cmake -GNinja ..`

`ninja` to build GMT
or 
`ninja docs_html` to build the Documentation

`ninja install` to install whatever has been build

I observed `ninja docs_html` runing the same in a fresh and in a populated `build/` directory. And it takes about the same time compared to `cmake`. No real speed increase there? Compared to @joa-quim 30s build time. `Ninja` apparently does the same image generation as `cmake` does:

```
kristof build % ninja docs_html
[59/212] Generating ../rst/source/_images/images/GMT_autolegend.png
```

I have the feeling I'm doing something wrong/different – but I haven't figured out where my mistake is.

**Comment by seisman** [comment]
> `Ninja` apparently does the same image generation as `cmake` does:
> 
> ```
> kristof build % ninja docs_html
> [59/212] Generating ../rst/source/_images/images/GMT_autolegend.png
> ```

Likely there is a bug (PNGs are always re-generated) when we migrated the baseline images to DVC. I will find some time to see if it's a bug.

**Comment by joa-quim** [comment]
My computer must be becoming lazier. It took 
- 65 s for the step ``ninja`` (no install just compile all sources)
- 3 min 30 s for first time ``ninja docs_html``
- 55 sec for a second run of ``ninja docs_html``

And yes, the ``Generating ../rst/source/_images/images/...`` is run every time.

**Comment by KristofKoch** [comment]
Our times are pretty similar @joa-quim. The ratios are consistent. I estimate your machine has fewer threads but a higher single thread performance than mine. 

Execution times in a fresh `build` directory:

command | time
-- | --
`PARALLEL_LEVEL=12 cmake -GNinja ..` | 0:35 min
`ninja` | 0:55 min 
`ninja docs_html` _(first run)_ | 3:07 min
`ninja docs_html` _(subsequent runs)_  | 1:14 min

**Comment by KristofKoch** [comment]
> I will find some time to see if it's a bug.

Hi @seisman, were you able to look into it?

**Comment by seisman** [comment]
Reopening the issue because there are still some features that need to be implemented.

**Comment by seisman** [comment]
> * All plots are generated from scratch every time. That is quite time consuming and for the most part unnecessary when I'm just fixing links, typos, layout, etc. Sure, `--parallel` does help but still wasted time.

The issue has been fixed in https://github.com/GenericMappingTools/gmt/pull/7340.

> * No apparent way of installing just the documentation. Not really necessary to install everything (gmt executable, library, development headers and built-in data) every time. Not as time consuming as building but still wasted time.

I'm reading this post (https://stackoverflow.com/a/9192877). It seems there are two ways to install documentation only:
```
cmake -DCOMPONENT=Documentation -P cmake_install.cmake
```
or 
```
cmake --install . --component Documentation  # requires CMake>=3.15
```
In GMT, valid components include `Runtime`, `Documentation`, `GSHHG` and `DCW`.

I'm wondering if we should document it in BUILDING.md.
